### PR TITLE
UCP/EP: Allocate endpoint using a strided allocator

### PR DESCRIPTION
**Pull request by yosefe**

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6706/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6722/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6725/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
@evgeny-leksikov pls take a look

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6729/ for details (Mellanox internal link).

File: src/ucp/core/ucp_worker.c [hunk 1]
```diff
@@ -435,18 +428,8 @@ ucp_worker_iface_error_handler(void *arg, uct_ep_h uct_ep, ucs_status_t status)
...
-        ucs_error("Error %s was not handled for ep %p - "
-                  UCT_TL_RESOURCE_DESC_FMT,
-                  ucs_status_string(status), ucp_ep,
-                  UCT_TL_RESOURCE_DESC_ARG(tl_rsc));
-        return status;
-    }
-
-    err_cb = kh_val(&worker->ep_errh_hash, ucp_ep_errh_iter);
-    err_cb(ucp_ep->user_data, ucp_ep, status);
+    ucp_ep_ext_gen(ucp_ep)->err_cb(ucp_ep_ext_gen(ucp_ep)->user_data, ucp_ep,
```
Comment (evgeny-leksikov) [inline src/ucp/core/ucp_worker.c]: Seems like `err_cb` can be uninitialized here if user doesn't pass UCP_EP_PARAM_FIELD_ERR_HANDLER

Comment (yosefe) [reply src/ucp/core/ucp_worker.c]: good catch, fixed

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6733/ for details (Mellanox internal link).

**Pull request merged**